src/shared_memory_manager.cc

```diff
@@ -26,22 +26,199 @@
 
 #include "shared_memory_manager.h"
 
+#include "common.h"
+#include "triton/common/logging.h"
+
 // Not supporting shared memory for now
-#ifdef _WIN32
+#ifndef _WIN32
+#include <errno.h>
+#include <fcntl.h>
+#include <sys/mman.h>
+#include <unistd.h>
+#endif
+
 namespace triton { namespace server {
-SharedMemoryManager::~SharedMemoryManager() {}
+
+#ifdef _WIN32
+namespace {
 
 TRITONSERVER_Error*
-SharedMemoryManager::RegisterSystemSharedMemory(
-    const std::string& name, const std::string& shm_key, const size_t offset,
-    const size_t byte_size)
+OpenSharedMemoryRegion(
+    const std::string& shm_key, const size_t byte_size, HANDLE* shm_handle)
+{
+  // The CreateFileMapping function takes a high-order and low-order DWORD (4
+  // bytes each) for size. 'size_t' can either be 4 or 8 bytes depending on the
+  // operating system. To handle both cases agnostically, we cast 'byte_size' to
+  // uint64 to ensure we have a known size and enough space to perform our
+  // logical operations.
+  uint64_t upperbound_size = (uint64_t)byte_size;
+  DWORD high_order_size = (upperbound_size >> 32) & 0xFFFFFFFF;
+  DWORD low_order_size = upperbound_size & 0xFFFFFFFF;
+
+  *shm_handle = OpenFileMapping(
+      FILE_MAP_ALL_ACCESS,  // read/write access
+      FALSE,                // cannot inherit handle
+      shm_key.c_str());     // name of mapping object
+
+  if (*shm_handle == NULL) {
+    LOG_VERBOSE(1) << "OpenFileMapping failed with error code: "
+                   << std::to_string(GetLastError());
```

Comment:
This goes for all instances: I'd like to not just get the numerical error code back from windows, I'd like to see the message which it translates to as well. Please see [shared_library.cc](https://github.com/triton-inference-server/core/blob/main/src/shared_library.cc#L68-L79) in the core repo for ideas.  We should probably wrap this type of error handling in a function to make this easier to use.

Reply:
This is a great idea! And given that Windows error handling will now be present in several repos, I'm wondering if it makes sense to extend Triton's general error handling to accept Windows error codes and convert them for this purpose.

Reply:
New error will look something like this:
```
OpenFileMapping failed with error code: 2 - The system cannot find the file specified.
```